LUMO_toolbox/DOTHUB_LUMO2nirs.m

```diff
@@ -331,6 +332,24 @@
 % Define AUX matrix #######################################################
 aux = zeros(length(t),8);
 
+% Convert zero intensity values to noise floor estimate ###################
+if size(d,1) == 1 % i.e., if only one sample
+    mnD = d;
+else
+    mnD = mean(d);
+end
+dists_3D = DOTHUB_getSDdists(SD3D);
+SDS_noise = 70; % SD channels greater than this (mm) are deemed to be in the the noise floor
+if sum(d==0,'all') > 0
+    if max(dists_3D) >= SDS_noise
+        noisefloorest = mean(mnD(dists_3D>SDS_noise));
+        d(d == 0) = noisefloorest;
+        disp('Warning - noise floor estimate has been assigned to zero intensity values')
+    else
+        disp('Warning - intensity data contain zero values')
```

Comment:
Why not give this warning and then set d(d==0) = 1e-6 or something?  It will fuck up the data, of course, but it's probably better that than leaving zeros in it.  I guess the question is whether we want to hide that these zeros ever exist or whether the goal is just to allow them to be processed, even if the resulting channels are screwed up.

Reply:
Ok - if max SDS is less than 70 mm, then I've set the zero values to be 1e-6. I think we should give a warning in both cases and alert the user. Ultimately this is something that will be reoslved in firmwre or software soon hopefully.